### ISSUE TITLE
RandomRDF throws ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by denkv** [dice-group/ldcbench#19]
With `2` nodes and average degree of `4`.

https://travis-ci.org/dice-group/ldcbench/builds/489660277

If such values should not be used it should probably throw `IllegalArgumentException`.

**Comment by MichaelRoeder** [comment]
@abdoudesouki please add a check for the input variables when the method for the graph generation is called.

**Comment by abdoudesouki** [comment]
fixed

**Comment by MichaelRoeder** [comment]
I think that this is not fully correct :thinking: 
Shouldn't it be `(degree >= N)`?

**Comment by MichaelRoeder** [comment]
No, it is a directed graph... so the maximum degree of a node can be `2 * (N - 1)` if we assume that we have only one type of edge :thinking: 
Depending on your implementation of Barabasi, you should choose one of these boundaries.
